### DISCUSSION TITLE
Creat Temporary Password

### DISCUSSION THREAD

**Discussion by igoresanto** [jasonacox/tinytuya#434] [Q&A]
Hello

Is it possible to create a temporary password on a Tuya Zigbee 3.4 smart lock locally?

I was able to remotely open and request registration on the lock, but I was unable to create a previously defined password or register an already known ID card.

I'm creating a control system for several airbnb apartments, but we have some problems with the internet, so we want to do it locally.

Thank you very much for your attention.

**Comment by uzlonewolf** [comment]
Unfortunately I'm not familiar with these devices so I can't offer many suggestions.  Do the device logs on iot.tuya.com show anything useful after setting a password through the app?

Do you have a link for somewhere selling these locks?

**Comment by igoresanto** [comment]
When I create it through the app the log appears like this:
![Captura de tela 2023-12-12 223902](https://github.com/jasonacox/tinytuya/assets/5455788/8135dbee-d6f5-41b8-9116-e59ab0a60607)

Link:
https://pt.aliexpress.com/item/1005006010257695.html?spm=a2g0o.productlist.main.33.50456aef9MlUrS&algo_pvid=d6b8a83e-cca9-4415-b5e5-df1b92488d9e&algo_exp_id=d6b8a83e-cca9-4415-b5e5-df1b92488d9e-16&pdp_npi=4%40dis%21BRL%21807.25%21251.95%21%21%211115.60%21%21%402103247117024316246158758e1367%2112000035308467980%21sea%21BR%210%21AB&curPageLogUid=XPbjDLQaLie5

    {
        "name": "EKAZA Fechadura Digital Zigbee",
        "id": "ebc6ad2a0e860bb025aw2j",
        "key": "zfcctcR@Jr&R6wR1",
        "mac": "04:0d:84:ff:fe:61:73:8a",
        "uuid": "040d84fffe61738a",
        "sn": "10009494600160",
        "category": "jtmspro",
        "product_name": "EKAZA Door locker Zigbee",
        "product_id": "nxma1uam",
        "biz_type": 0,
        "model": "EKAV-T229Z",
        "sub": true,
        "icon": "https://images.tuyaus.com/smart/icon/ay1540090774726g5666/783633fb7be0f70d8dea625a19ca03c0.png",
        "node_id": "040d84fffe61738a",
        "mapping": {
            "1": {
                "code": "unlock_fingerprint",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "2": {
                "code": "unlock_password",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "3": {
                "code": "unlock_temporary",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "4": {
                "code": "unlock_dynamic",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "5": {
                "code": "unlock_card",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "8": {
                "code": "open_close",
                "type": "Boolean",
                "values": {}
            },
            "9": {
                "code": "alarm_lock",
                "type": "Enum",
                "values": {
                    "range": [
                        "wrong_finger",
                        "wrong_password",
                        "wrong_card",
                        "wrong_face",
                        "tongue_bad",
                        "too_hot",
                        "tongue_not_out",
                        "pry",
                        "key_in",
                        "power_off",
                        "shock",
                        "unclosed_time",
                        "low_battery",
                        "defense"
                    ]
                }
            },
            "10": {
                "code": "residual_electricity",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 100,
                    "scale": 0,
                    "step": 1
                }
            },
            "12": {
                "code": "reverse_lock",
                "type": "Boolean",
                "values": {}
            },
            "14": {
                "code": "doorbell",
                "type": "Boolean",
                "values": {}
            },
            "15": {
                "code": "doorbell_song",
                "type": "Enum",
                "values": {
                    "range": [
                        "ding_0",
                        "ding_1",
                        "ding_2",
                        "ding_3",
                        "ding_4",
                        "ding_5",
                        "ding_6",
                        "ding_7",
                        "ding_8",
                        "ding_9"
                    ]
                }
            },
            "22": {
                "code": "remote_result",
                "type": "Boolean",
                "values": {}
            },
            "35": {
                "code": "hijack",
                "type": "Boolean",
                "values": {}
            },
            "41": {
                "code": "unlock_remote",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "47": {
                "code": "closed_opened",
                "type": "Enum",
                "values": {
                    "range": [
                        "open",
                        "closed",
                        "unknown"
                    ]
                }
            },
            "48": {
                "code": "remote_no_pd_setkey",
                "type": "Raw",
                "values": {}
            },
            "49": {
                "code": "remote_no_dp_key",
                "type": "Raw",
                "values": {}
            },
            "54": {
                "code": "unlock_method_create",
                "type": "Raw",
                "values": {}
            },
            "55": {
                "code": "unlock_method_delete",
                "type": "Raw",
                "values": {}
            },
            "57": {
                "code": "lock_motor_state",
                "type": "Boolean",
                "values": {}
            },
            "62": {
                "code": "unlock_voice_remote",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "68": {
                "code": "unlock_offline_pd",
                "type": "Raw",
                "values": {}
            }
        },
        "parent": "",
        "ip": "",
        "version": ""
    },
    {
        "name": "EKAZA Hub Zigbee Multi Mode",
        "id": "eb2452e7ea155291ddap2z",
        "key": "zfcctcR@Jr&R6wR1",
        "mac": "50:8b:b9:9e:9a:b1",
        "uuid": "f2ca98a24c05deb0",
        "sn": "10011241600183",
        "category": "wg2",
        "product_name": "EKAZA Hub Zigbee Multi Mode",
        "product_id": "p9afxlaheycyjt62",
        "biz_type": 0,
        "model": "EKAT-T3010M",
        "sub": true,
        "icon": "https://images.tuyaus.com/smart/icon/ay1540090774726g5666/e276d66822f95dcae923baea5176bd3c.jpg",
        "mapping": {
            "4": {
                "code": "switch_alarm_sound",
                "type": "Boolean",
                "values": {}
            },
            "32": {
                "code": "master_state",
                "type": "Enum",
                "values": {
                    "range": [
                        "normal",
                        "alarm"
                    ]
                }
            },
            "34": {
                "code": "factory_reset",
                "type": "Boolean",
                "values": {}
            },
            "45": {
                "code": "alarm_active",
                "type": "String",
                "values": "{\"maxlen\":255}"
            }
        },
        "parent": "",
        "ip": "192.168.0.16",
        "version": "3.4"
    }

**Comment by igoresanto** [comment]
I even found some material here:

https://developer.tuya.com/en/docs/iot/zigbee-apartment-doorlock-dp-document?id=K9fx87lrdsqwb#title-4-Create%20temporary%20passwords

is here:

https://developer.tuya.com/en/docs/iot/zigbee-apartment-doorlock-developer?id=K9fekzk7k8ca2#title-22-Status%20description

But you can't use

**Comment by uzlonewolf** [comment]
I'm assuming it packs it into DP 54 somehow.  I have one of these devices coming so I can have a look at it, hopefully it will be here later this week.

**Comment by igoresanto** [comment]
I will certainly be here and I hope I can contribute to this process

**Comment by uzlonewolf** [comment]
I just discovered https://developer.tuya.com/en/docs/iot/zigbee-doorlock-dp?id=K9fembhbeab0p which says "Create temporary passwords" should be DP 24.  My guess is that DP exists but is just not being included in the JSON, possibly because of https://github.com/jasonacox/tinytuya/blob/master/DP_Mapping.md .  What happens if you send the lock
```python
d.set_value(24, 'A+cAAQAAZXeZp2V46yoAAAAAAAAAOTk5OTk5')
```
?

**Comment by uzlonewolf** [comment]
I didn't notice when I bought it, but that one you linked is Bluetooth (not Zigbee) and uses a completely different DP set.  I can't seem to find any Zigbee locks from from EKAZA, and the other brands are twice the price of the Bluetooth one.  As such I'm not going to be able to do much besides offer suggestions to try.

**Reply by uzlonewolf** [reply]
The DPs for the Bluetooth one linked above:

```json
    {
        "name": "Smart Lock",
        "id": "eb1****e",
        "key": "ad****7f",
        "mac": "DC:****:0C",
        "uuid": "a7****a4",
        "category": "jtmspro",
        "product_name": "Smart Lock",
        "product_id": "xicdxood",
        "biz_type": 18,
        "model": "C502",
        "sub": true,
        "icon": "https://images.tuyaus.com/smart/program_category_icon/jtmspro.png",
        "node_id": "a7****a4",
        "mapping": {
            "1": {
                "code": "unlock_method_create",
                "type": "Raw",
                "values": {}
            },
            "2": {
                "code": "unlock_method_delete",
                "type": "Raw",
                "values": {}
            },
            "3": {
                "code": "unlock_method_modify",
                "type": "Raw",
                "values": {}
            },
            "8": {
                "code": "residual_electricity",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 100,
                    "scale": 0,
                    "step": 1
                }
            },
            "12": {
                "code": "unlock_fingerprint",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "13": {
                "code": "unlock_password",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "14": {
                "code": "unlock_dynamic",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "15": {
                "code": "unlock_card",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "19": {
                "code": "unlock_ble",
                "type": "Integer",
                "values": {
                    "min": 0,
                    "max": 999,
                    "scale": 0,
                    "step": 1
                }
            },
            "21": {
                "code": "alarm_lock",
                "type": "Enum",
                "values": {
                    "range": [
                        "wrong_finger",
                        "wrong_password",
                        "wrong_card",
                        "wrong_face",
                        "tongue_bad",
                        "too_hot",
                        "unclosed_time",
                        "tongue_not_out",
                        "pry",
                        "key_in",
                        "low_battery",
                        "power_off",
                        "shock",
                        "defense"
                    ]
                }
            },
            "24": {
                "code": "doorbell",
                "type": "Boolean",
                "values": {}
            },
            "28": {
                "code": "language",
                "type": "Enum",
                "values": {
                    "range": [
                        "chinese_simplified",
                        "english"
                    ]
                }
            },
            "31": {
                "code": "beep_volume",
                "type": "Enum",
                "values": {
                    "range": [
                        "mute",
                        "low",
                        "normal",
                        "high"
                    ]
                }
            },
            "44": {
                "code": "rtc_lock",
                "type": "Boolean",
                "values": {}
            },
            "54": {
                "code": "synch_method",
                "type": "Raw",
                "values": {}
            },
            "70": {
                "code": "check_code_set",
                "type": "Raw",
                "values": {}
            },
            "71": {
                "code": "ble_unlock_check",
                "type": "Raw",
                "values": {}
            }
        },
        "parent": ""
    }
```
